### PR TITLE
Added min y level setting

### LINKED ISSUE #463: Fix World Y min
https://github.com/NetherGamesMC/PocketMine-MP/blob/2398a0ec9255de1c8188bff0e23ebd61ec834a2e/src/entity/Entity.php#L659

This has been fixed in pmmp - https://github.com/pmmp/PocketMine-MP/blob/9b6a0c99450949c4350f5d3bf49396c19e6ce79c/src/entity/Entity.php#L657

### PR DESCRIPTION
**Pull request by Superomarking**
As requested by dries, Fixes #463 and has been tested

### CODE REVIEW

File: src/entity/Entity.php [hunk 1]
```diff
@@ -656,7 +657,10 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 		}
 		$this->checkBlockIntersectionsNextTick = true;
 
-		if($this->location->y <= 0 && $this->isAlive()){
+		$cfg = $this->server->getConfigGroup();
+		$minY = $cfg->getPropertyInt(YmlServerProperties::LEVEL_SETTINGS_MIN_Y, 0);
+
+		if($this->location->y <= $minY && $this->isAlive()){
```
Comment (dries-c) [inline src/entity/Entity.php]: Given this method will get called often it might be nice to load the minY into the world and using $world->getMinY() or something.

**Review by dries-c — CHANGES_REQUESTED** [review]

File: src/entity/Entity.php (hunk 1 above)
Comment (Superomarking) [reply src/entity/Entity.php]: I'll make it settable and public for developers to use if they need.

Comment (Superomarking) [reply src/entity/Entity.php]: Now that I think about it, I won't make it settable 😔

File: src/entity/Entity.php [hunk 2]
```diff
@@ -656,7 +656,7 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 		}
 		$this->checkBlockIntersectionsNextTick = true;
 
-		if($this->location->y <= 0 && $this->isAlive()){
+		if($this->location->y <= $this->getWorld()->getMinVoidY() && $this->isAlive()){
```
Comment (dries-c) [inline src/entity/Entity.php]: getMinVoidY sounds like a pleonasm

File: src/world/World.php [hunk 3]
```diff
@@ -499,10 +500,13 @@ public function __construct(
...
 		$this->logger = new \PrefixedLogger($server->getLogger(), "World: $this->displayName");
 
+        $cfg = $this->server->getConfigGroup();
+
 		$this->minY = $this->provider->getWorldMinY();
 		$this->maxY = $this->provider->getWorldMaxY();
+        $this->minVoidY = $cfg->getPropertyInt(YmlServerProperties::LEVEL_SETTINGS_MIN_Y, 0);
 
-		$this->server->getLogger()->info($this->server->getLanguage()->translate(KnownTranslationFactory::pocketmine_level_preparing($this->displayName)));
+        $this->server->getLogger()->info($this->server->getLanguage()->translate(KnownTranslationFactory::pocketmine_level_preparing($this->displayName)));
```
Comment (dries-c) [inline src/world/World.php]: unrelated change

**Review by dries-c — CHANGES_REQUESTED** [review]

File: src/entity/Entity.php (hunk 2 above)
Comment (Superomarking) [reply src/entity/Entity.php]: getDamageY sounds fine?

**Comment by dries-c** [comment]
The codestyle CI didn't approve

**Review by Superomarking — COMMENTED** [review]
I've been using spaces instead of tabs, will fix

**Comment by dries-c** [comment]
One tiny CS error left
